synfig-core/src/synfig/timepointcollect.cpp

```diff
@@ -88,7 +89,7 @@ synfig::waypoint_collect(std::set<Waypoint, std::less<UniqueID> >	&waypoint_set,
 		Layer::DynamicParamList::const_iterator iter;
 		int ret(0);
 		etl::handle<Layer_PasteCanvas> p = etl::handle<Layer_PasteCanvas>::cast_dynamic(layer);
-		if (!p){
+        if ((!p && ignore_dynamic_parameters) || !ignore_dynamic_parameters){
```

Comment:
This can be simplified by a absorption law in Boolean Algebra:
https://www.geeksforgeeks.org/properties-of-boolean-algebra/
https://www.allaboutcircuits.com/textbook/digital/chpt-7/boolean-rules-for-simplification/

Indentation is wrong: use Tabs.

Reply:
nice, will do